programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -56,9 +70,26 @@ pub mod gateway {
         Ok(())
     }
 
+    pub fn update_authority(
```

Comment:
Can we add some comments what is the usage of the authority address?

Reply:
i'm going to merge this one, and start another one that includes docs, and logs.